apps/api/src/services/db/statsService.ts

```diff
@@ -15,66 +15,107 @@ type GraphData = {
-  const secondCompareBlockStats = await Block.findOne({
-    order: [["datetime", "ASC"]],
-    where: {
-      datetime: { [Op.gte]: secondCompareDate }
-    }
-  });
-
-  return {
-    now: {
-      date: latestBlockStats.datetime,
-      height: latestBlockStats.height,
-      activeLeaseCount: latestBlockStats.activeLeaseCount,
-      totalLeaseCount: latestBlockStats.totalLeaseCount,
-      dailyLeaseCount: latestBlockStats.totalLeaseCount - compareBlockStats.totalLeaseCount,
-      totalUAktSpent: latestBlockStats.totalUAktSpent,
-      dailyUAktSpent: latestBlockStats.totalUAktSpent - compareBlockStats.totalUAktSpent,
-      totalUUsdcSpent: latestBlockStats.totalUUsdcSpent,
-      dailyUUsdcSpent: latestBlockStats.totalUUsdcSpent - compareBlockStats.totalUUsdcSpent,
-      totalUUsdSpent: latestBlockStats.totalUUsdSpent,
-      dailyUUsdSpent: latestBlockStats.totalUUsdSpent - compareBlockStats.totalUUsdSpent,
-      activeCPU: latestBlockStats.activeCPU,
-      activeGPU: latestBlockStats.activeGPU,
-      activeMemory: latestBlockStats.activeMemory,
-      activeStorage: latestBlockStats.activeEphemeralStorage + latestBlockStats.activePersistentStorage
-    },
-    compare: {
-      date: compareBlockStats.datetime,
-      height: compareBlockStats.height,
-      activeLeaseCount: compareBlockStats.activeLeaseCount,
-      totalLeaseCount: compareBlockStats.totalLeaseCount,
-      dailyLeaseCount: compareBlockStats.totalLeaseCount - secondCompareBlockStats.totalLeaseCount,
-      totalUAktSpent: compareBlockStats.totalUAktSpent,
-      dailyUAktSpent: compareBlockStats.totalUAktSpent - secondCompareBlockStats.totalUAktSpent,
-      totalUUsdcSpent: compareBlockStats.totalUUsdcSpent,
-      dailyUUsdcSpent: compareBlockStats.totalUUsdcSpent - secondCompareBlockStats.totalUUsdcSpent,
-      totalUUsdSpent: compareBlockStats.totalUUsdSpent,
-      dailyUUsdSpent: compareBlockStats.totalUUsdSpent - secondCompareBlockStats.totalUUsdSpent,
-      activeCPU: compareBlockStats.activeCPU,
-      activeGPU: compareBlockStats.activeGPU,
-      activeMemory: compareBlockStats.activeMemory,
-      activeStorage: compareBlockStats.activeEphemeralStorage + compareBlockStats.activePersistentStorage
-    }
-  };
+  try {
+    const latestBlockStats = await Block.findOne({
+      where: {
+        isProcessed: true,
+        totalUUsdSpent: { [Op.not]: null }
+      },
+      order: [["height", "DESC"]]
+    });
+  
+    const compareDate = subHours(latestBlockStats.datetime, 24);
+    const compareBlockStats = await Block.findOne({
+      order: [["datetime", "ASC"]],
+      where: {
+        datetime: { [Op.gte]: compareDate }
+      }
+    });
+  
+    const secondCompareDate = subHours(latestBlockStats.datetime, 48);
+    const secondCompareBlockStats = await Block.findOne({
+      order: [["datetime", "ASC"]],
+      where: {
+        datetime: { [Op.gte]: secondCompareDate }
+      }
+    });
+  
+    return {
+      now: {
```

Comment:
I think props `prev`/`next`, `prev`/`current` would better describe the contents. `compare` makes me wonder what that might be, it's also a verb which complicates it more. `now` sounds like a timestamp value, sounds confusing.

Reply:
I agree `prev`/`current` would be better names. This is part of the public api however, maybe we should hold off renaming those if it's not needed in case somebody uses it? At least until we release a v2 of the api.

Reply:
Yea, agreed. This would be a breaking change, should be versioned.

Reply:
I missed these props were here before. Scratch my comment )

Reply:
We can't know for sure. This kind of changes should be versioned